apollo-router/src/plugins/coprocessor/supergraph.rs

```diff
@@ -0,0 +1,881 @@
+                let new_deferred_response: crate::response::Response =
+                    match co_processor_output.body {
+                        Some(value) => serde_json::from_value(value)?,
+                        None => deferred_response,
+                    };
+
+                if let Some(context) = co_processor_output.context {
+                    for (key, value) in context.try_into_iter()? {
+                        generator_map_context.upsert_json_value(key, move |_current| value);
+                    }
+                }
+
+                // We return the deferred_response into our stream of response chunks
+                Ok(new_deferred_response)
+            }
+        })
+        .map(|res: Result<response::Response, BoxError>| match res {
+            Ok(response) => response,
+            Err(e) => {
+                tracing::error!("coprocessor error handling deferred supergraph response: {e}");
+                response::Response::builder()
+                    .error(
+                        Error::builder()
+                            .message("Internal error handling deferred response")
+                            .extension_code("INTERNAL_ERROR")
+                            .build(),
+                    )
+                    .build()
+            }
+        });
+
+    //response.response = http::Response::from_parts(parts, stream::once(future::ready(new_body)));
+
+    // Create our response stream which consists of the bytes from our first body chained with the
+    // rest of the responses in our mapped stream.
+    //let bytes = hyper::body::to_bytes(body).await.map_err(BoxError::from);
+    let stream = once(ready(new_body)).chain(mapped_stream).boxed();
+
+    // Finally, return a response which has a Body that wraps our stream of response chunks.
+    Ok(supergraph::Response {
+        context,
+        response: http::Response::from_parts(parts, stream),
+    })
+}
+
+#[cfg(test)]
+mod tests {
+    use std::sync::Arc;
+
+    use futures::future::BoxFuture;
+    use http::StatusCode;
+    use hyper::Body;
+    use serde_json::json;
+    use tower::BoxError;
+    use tower::ServiceExt;
+
+    use super::super::*;
+    use super::*;
+    use crate::plugin::test::MockHttpClientService;
+    use crate::plugin::test::MockSupergraphService;
+    use crate::services::supergraph;
+
+    #[allow(clippy::type_complexity)]
+    pub(crate) fn mock_with_callback(
+        callback: fn(
+            hyper::Request<Body>,
+        ) -> BoxFuture<'static, Result<hyper::Response<Body>, BoxError>>,
+    ) -> MockHttpClientService {
+        println!("mock_with_callback [{}]", line!());
+        let mut mock_http_client = MockHttpClientService::new();
+        mock_http_client.expect_clone().returning(move || {
+            println!("mock_with_callback [{}]", line!());
+
+            let mut mock_http_client = MockHttpClientService::new();
+            //mock_http_client.expect_call().returning(callback);
```

Comment:
ditto

Reply:
fixed in [2680e59](https://github.com/apollographql/router/pull/3408/commits/2680e5933309b59c47753b58eb1c9b4b80d254f1)